crates/goose/src/agents/code_execution_extension.rs

```diff
@@ -82,19 +82,31 @@ impl ToolInfo {
         let (server_name, tool_name) = tool.name.as_ref().split_once("__")?;
         let param_names = get_parameter_names(tool);
 
-        let schema: InputSchema =
-            serde_json::from_value(Value::Object(tool.input_schema.as_ref().clone()))
-                .unwrap_or_default();
+        let mut schema_value = Value::Object(tool.input_schema.as_ref().clone());
+        let _ = unbinder::dereference_schema(&mut schema_value, unbinder::Options::default());
+        let schema: InputSchema = serde_json::from_value(schema_value).unwrap_or_default();
 
         let params = param_names
             .iter()
             .map(|name| {
                 let ty = schema
                     .properties
                     .get(name)
-                    .and_then(|p| p.get("type"))
-                    .and_then(|t| t.as_str())
-                    .unwrap_or("any");
+                    .and_then(|p| {
+                        // Check for oneOf with const (schemars generates this for enums)
```

Comment:
if this comment is too much, happy to scrub it. This was the key issue in WebScrape not working, and it was getting gnarly to handle these by hand. Also, there are other cases besides oneOf we probably need to handle soon. Just didn't want to go too much in one PR